contracts/clients/src/BlsProvider.ts

```diff
@@ -9,7 +9,7 @@ import poll from "./helpers/poll";
 export default class BlsProvider extends ethers.providers.JsonRpcProvider {
```

Comment:
This class has two todos that are tagged to be resolved as part of this issue/PR:
- https://github.com/web3well/bls-wallet/blob/8ef89face77a7b8cf7619f46d3cf6aafbab7551a/contracts/clients/src/BlsProvider.ts#L33
- https://github.com/web3well/bls-wallet/blob/8ef89face77a7b8cf7619f46d3cf6aafbab7551a/contracts/clients/src/BlsProvider.ts#L50

I could remove these checks and run `this.getSigner(privateKey)`, but that would require passing the private key to each of these methods. 

The original motivation to removing these checks, and subsequently the signer property on the BlsProvider class, was because the ethers JsonRpcProvider has no such reference to it's signer. Maybe such a difference from the ethers class is ok, but then again, we are trying to maintain a very similar interface. Open to suggestions on making this cleaner, or whether it's ok to be left as is.